src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/AvoidConstArrays.cs

```diff
@@ -0,0 +1,132 @@
+
+                    if (context.Operation is IArrayCreationOperation arrayCreationOperation) // For arrays passed as arguments
+                    {
+                        argumentOperation = arrayCreationOperation.GetAncestor<IArgumentOperation>(OperationKind.Argument);
+                        if (argumentOperation is null || argumentOperation.Parameter.IsParams)
+                        {
+                            return;
+                        }
+                    }
+                    else if (context.Operation is IInvocationOperation invocationOperation) // For arrays passed in extension methods, like in LINQ
+                    {
+                        IEnumerable<IOperation> invocationDescendants = invocationOperation.Descendants();
+                        if (invocationDescendants.Any(x => x is IArrayCreationOperation)
+                            && invocationDescendants.Any(x => x is IArgumentOperation))
+                        {
+                            // This is an invocation that contains an array as an argument
+                            // This will get caught by the first case in another cycle
+                            return;
+                        }
+
+                        argumentOperation = invocationOperation.Arguments.FirstOrDefault();
+                        if (argumentOperation is not null)
+                        {
+                            if (argumentOperation.Children.First() is not IConversionOperation conversionOperation
+                                || conversionOperation.Operand is not IArrayCreationOperation arrayCreation)
+                            {
+                                return;
+                            }
+                            arrayCreationOperation = arrayCreation;
+                        }
+                        else // An invocation, extension or regular, has an argument, unless it's a VB extension method call
+                        {
+                            // For VB extension method invocations, find a matching child
+                            arrayCreationOperation = (IArrayCreationOperation)invocationOperation.Descendants()
+                                .FirstOrDefault(x => x is IArrayCreationOperation);
+                            if (arrayCreationOperation is null)
+                            {
+                                return;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        return;
+                    }
+
+                    if (argumentOperation is not null)
+                    {
+                        ITypeSymbol originalDefinition = argumentOperation.Parameter.Type.OriginalDefinition;
+
+                        // Can't be a ReadOnlySpan, as those are already optimized
+                        if (readonlySpanType is not null && originalDefinition.Equals(readonlySpanType))
```

Comment:
You probably want to use `SymbolEqualityComparer` for the symbols equality.

```suggestion
                        if (SymbolEqualityComparer.Default.Equals(readonlySpanType, originalDefinition))
```